engine/src/main/java/com/datatorrent/stram/codec/DefaultStatefulStreamCodec.java

```diff
@@ -63,10 +64,11 @@ public DefaultStatefulStreamCodec()
     classResolver = (ClassResolver)getClassResolver();
     this.pairs = classResolver.pairs;
     classResolver.init();
+    dataStatePair = new DataStatePair();
```

Comment:
The only assumption made is that the caller is responsible for processing returned pair prior to calling toDataStatePair() again. This is exactly how toDataStatePair() is used.